typings.d.ts

```diff
@@ -72,11 +71,35 @@ declare module 'resin-components' {
 	class RenderableElementWithProps<PropsType, StateType> extends Component<
 		PropsType,
 		StateType
-	> {
-		refs: any;
-		render(): JSX.Element | null;
+	> {}
+
+	interface Coloring {
+		primary?: boolean;
+		secondary?: boolean;
+		tertiary?: boolean;
+		quartenary?: boolean;
+	}
+
+	interface BannerProps extends DefaultProps {
+		backgroundImage?: string;
+	}
+
+	class BannerProps extends RenderableElementWithProps<BoxProps, any> {}
+
+	interface BoxProps extends DefaultProps {
+		flex?: string | string[];
+		order?: number | string | Array<number | string>;
 	}
 
+	class Box extends RenderableElementWithProps<BoxProps, any> {}
+
+	interface ButtonProps extends DefaultProps, Coloring {
+		square?: boolean;
+		disabled?: boolean;
```

Comment:
I think that this is missing the `emphasized`, `outline` & `underline` properties.